Source/WebCore/platform/graphics/transforms/TransformOperations.cpp

```diff
@@ -106,7 +106,7 @@ TransformOperations TransformOperations::blend(const TransformOperations& from,
     for (unsigned i = 0; i < maxOperationCount; i++) {
         RefPtr<TransformOperation> fromOperation = (i < fromOperationCount) ? from.operations()[i].get() : nullptr;
         RefPtr<TransformOperation> toOperation = (i < toOperationCount) ? operations()[i].get() : nullptr;
-        if ((prefixLength && i >= *prefixLength) || (fromOperation && toOperation && !fromOperation->sharedPrimitiveType(toOperation.get()))) {
+        if ((prefixLength && i >= *prefixLength && *prefixLength) || (fromOperation && toOperation && !fromOperation->sharedPrimitiveType(toOperation.get()))) {
```

Comment:
This seems wrong to me. There are other cases where the prefix might be empty as well such as well the operation lists do not match at all. I think in this case where one list is empty the prefix should be as long as the non-empty list and have compatible primitives. The code that handles this is in `SharedPrimitivesPrefix::update`:

```c++
        // If we haven't seen an operation at this index before, we can simply use our primitive type.
        if (i >= m_primitives.size()) {
            ASSERT(i == m_primitives.size());
            m_primitives.append(operation->primitiveType());
            continue;
        }
```

I suspect something else is going on and this might just be covering up the original issue.

Reply:
In the affected test, `SharedPrimitivesPrefix::update()` is never called due to the early return in `KeyframeEffect::checkForMatchingTransformFunctionLists()`:

```c++
void KeyframeEffect::checkForMatchingTransformFunctionLists()
{
    if (m_blendingKeyframes.size() < 2 || !m_blendingKeyframes.containsProperty(CSSPropertyTransform)) {
        m_transformFunctionListsMatchPrefix = 0;
        return;
    }

    SharedPrimitivesPrefix prefix;
    for (const auto& keyframe : m_blendingKeyframes)
        prefix.update(keyframe.style()->transform());

    m_transformFunctionListsMatchPrefix = prefix.primitives().size();
}
```

Indeed, in this case, there is a single keyframe.

Reply:
So my guess was to remove the `m_blendingKeyframes.size() < 2` clause which would only set `m_transformFunctionListsMatchPrefix` to `0` should there not be a single keyframe specifying a `transform`. This lets `SharedPrimitivesPrefix::update()` run and set the value to `1`.

But this regresses at least one test, `compositing/backing/backing-store-attachment-empty-keyframe.html`, which no longer produces a layer tree.

Reply:
What seems to me like a simple approach is to never attempt to do matrix-blending when either the `from` or `to` transform list is empty, not involving the prefix at all.

Reply:
This version of the change does look more reasonable, but I wonder...do things still work well if you simply change the line in question to this:

```c++
    if (m_blendingKeyframes.size() < 1 || !m_blendingKeyframes.containsProperty(CSSPropertyTransform)) {
        m_transformFunctionListsMatchPrefix = 0;
        return;
    }
```

Reply:
Also, if we don't have _any_ blending keyframes, then we won't blend at all, so the `m_transformFunctionListsMatchPrefix` value is really irrelevant.

Reply:
That said, to answer your question, your proposed change as an additional change on top of the current version of this patch regresses `compositing/backing/backing-store-attachment-empty-keyframe.html`.